### PR TITLE
Fix types in method signatures and docblocks

### LINKED ISSUE #70: moveMouseOver Method phpdoc saying only css selector string but accept string/array css selector
Hello,

Just used the module Webdriver module today and found than the phpdoc for **moveMouseOver** has for typehint

WebDriver.php: line 2804:
`
     * @param string $cssOrXPath css or xpath of the web element
`

But the example just above state than we can use it with the array selector `["css" => ".checkout"]`
I've tried running it with the array selector and it's working.

But my usage of `$I->moveMouseOver(["css" => ".checkout")` is marked as an phpstan error and could create other issue with other analysis tool.

### PR DESCRIPTION
**Pull request by Naktibalda**
I fixed many signatures that were stricter than necessary.

Closes #70

### CODE REVIEW

File: src/Codeception/Module/WebDriver.php [hunk 1]
```diff
@@ -1755,11 +1796,9 @@ public function unselectOption(WebDriverElement $select, $option): void
     }
 
     /**
-     * @param $context
-     * @param WebDriverElement|WebDriverBy $radioOrCheckbox
-     * @return mixed
+     * @param string|array|WebDriverBy|WebDriverElement $radioOrCheckbox
      */
-    protected function findCheckable($context, $radioOrCheckbox, bool $byValue = false)
+    protected function findCheckable($context, $radioOrCheckbox, bool $byValue = false): ?WebDriverElement
```
Comment (TavoNiievez) [inline src/Codeception/Module/WebDriver.php]: please add a typehint for `$context`

**Review by TavoNiievez — CHANGES_REQUESTED** [review]
Okay,
The changes that I would suggest in this PR have to do with the types of functions that are implemented from an interface.
Please verify that the signature of the methods corresponds exactly to each of the interfaces implemented in this class.
Taking a quick look I found for example that `loadSessionSnapshot` has a string `$name` parameter that is not typed yet, whereas you should remove all the `: void`s and `: ?string`s that you added in methods that implement an interface.
I am also curious about a matter. I didn't add a docblock in a function that implements an interface because I remember you once told me that doing that overwrites the documentation generated by `robo`. If that is the case, then you must also add the method description and the code example from the interface.

We should only change the typing of the methods that come from interfaces when we release a version that `points` and `requires` Codeception 5.
Even if I know that we are already breaking compatibility by modifying the other methods, we shouldn't mix up those changes. They must be different PR's as well as different versions.

**Comment by Naktibalda** [comment]
> Taking a quick look I found for example that `loadSessionSnapshot` has a string `$name` parameter that is not typed yet

I haven't typed it in my PR either.

The first build failed because I added type to parameter of `_closeSession` method.

> PHP Fatal error:  Declaration of Codeception\Module\WebDriver::_closeSession(?Facebook\WebDriver\Remote\RemoteWebDriver $webDriver = null): void must be compatible with Codeception\Lib\Interfaces\MultiSession::_closeSession($session = null)

https://github.com/Codeception/module-webdriver/runs/4633062856?check_suite_focus=true

so I removed it.

I guess that PHP does not complain about return type that is more specific than the interface.

**Comment by Naktibalda** [comment]
Good point about documentation, I will check if we can use `@inheritdoc`

**Comment by Naktibalda** [comment]
There is no inheritance, but I will try to implement it.

**Review by TavoNiievez — APPROVED** [review]
I'm not sure if this is ready for review again, but if you took my comments into account, it looks good to me this way.

File: src/Codeception/Module/WebDriver.php [hunk 2]
```diff
@@ -1676,7 +1685,7 @@ public function _loadSession($session)
      *
      * @api
      */
-    public function _backupSession()
+    public function _backupSession(): ?RemoteWebDriver
```
Comment (TavoNiievez) [inline src/Codeception/Module/WebDriver.php:1688]: @Naktibalda I just noticed that this method returns `: array` in Codeception 5
https://github.com/Codeception/Codeception/blob/a15b58781cbed2b3d5f21fd148cef474ee51101e/src/Codeception/Lib/Interfaces/MultiSession.php#L11
Should I remove the typing there?

File: src/Codeception/Module/WebDriver.php [hunk 3]
```diff
@@ -3108,36 +3132,36 @@ protected function matchFirstOrFail($page, $selector): WebDriverElement
      * $I->pressKey('#name', array('ctrl', 'a'), \Facebook\WebDriver\WebDriverKeys::DELETE); //=>''
      * ```
      *
-     * @param $char string|array Can be char or array with modifier. You can provide several chars.
+     * @param string|array|WebDriverBy $element
+     * @param array<string|string[]>$chars Can be char or array with modifier. You can provide several chars.
```
Comment (ThomasLandauer) [inline src/Codeception/Module/WebDriver.php:3136]: @Naktibalda According to this typehint, `$chars` always has to be an array; so the given example `$I->pressKey('#page', 'a');` would be invalid. Where did you take this from?

**Pull request merged**